### DISCUSSION TITLE
Frontend improvements: Component division and visual structure changes

### DISCUSSION THREAD

**Discussion by d4saher** [jyjblrd/Mocap-Drones#58] [Show and tell]
Hi there,

As I mentioned in my other discussion [here](https://github.com/jyjblrd/Low-Cost-Mocap/discussions/50), I'm having some trouble with the floor calibration process.

To better understand how the code works, I started splitting the frontend into views or panels to make it more readable. I created a new views folder that contains each panel as components:

- Camera Settings
- Camera Stream
- Control Panel
    · Control Drone View
    · Trajectory View
- Scene View

The chart is still in the App parent component.

In this version, Camera Settings, Control Drone View, and Trajectory View are on the same panel. I think you don't need to see them all at the same time, and it's more useful to view each one alongside the Scene View.
Here you can see some pictures of how it looks like:
![Captura desde 2024-07-18 14-27-33](https://github.com/user-attachments/assets/87fcd6c9-58b9-44e0-90a3-bbf3c6397f3b)
![Captura desde 2024-07-19 11-16-47](https://github.com/user-attachments/assets/6ae718af-c141-48c6-9d0a-f7eba669a0b3)

I'm not sure if the component division is good enought; it needs to be revised and tested as there are a lot of props being passed through components, and I might have missed something. Some things could be done differently to avoid that. For example, I don't think it's necessary to pass 'toWorldCoordsMatrix' or 'cameraPoses' to the backend each time you want to calculate something like origin, scale, or floor. It would be enough to store them on the backend and just update them each time they change. What do you think about this approach?

As always, the changes are on my [fork repository](https://github.com/d4saher/Low-Cost-Mocap.git) in the dev branch.

Looking forward to your feedback!

**Comment by jyjblrd** [comment]
Looks good. You are right, the front end just shows everything all at once and can be hard to navigate.

I pass stuff between the frontend and backend more than necessary to try keep the backend as stateless as possible. This is convenient as it means the backend and front end wont fall "out of sync", and since it is all running on the same computer we don't really need to optimize for bandwidth. That being said, there are probably some things which are passed too much.

**Comment by nbpilot007** [comment]
hi @d4saher 
I tried running your code with the new front end you made and re did the process, i face the same problem of one of my camera becoming the origin the front end button did work but i was not able to get them to the right position. Have you managed to get the system working? 

its been quite a few time since i am on it but have been stuck on the caliberating process.

**Reply by d4saher** [reply]
Hi @nbpilot007,

I didn't fix the acquire floor method; I just added some buttons to rotate the scene, allowing manual adjustments. Therefore, you can ignore the acquire floor step.

Unfortunately, I wasn't able to achieve a good camera pose calibration. I suspect I might be getting incorrect intrinsics or something else could be off.

I'll need to do some more testing.
